imi/src/test/java/org/motechproject/nms/imi/it/CdrFileServiceBundleIT.java

```diff
@@ -89,18 +111,76 @@ public void testServicePresent() {
 
 
     @Test
-    public void testParse() throws IOException, NoSuchAlgorithmException {
-        getLogger().debug("testParse()");
+    public void testVerify() throws IOException, NoSuchAlgorithmException {
+        getLogger().debug("testVerify()");
+
+        CdrHelper helper = new CdrHelper(settingsService, subscriptionService, subscriberDataService,
+                languageDataService, languageLocationDataService, circleDataService, stateDataService,
+                districtDataService, fileAuditRecordDataService);
+
+        helper.makeCdrs(1,1,1,1);
+        helper.makeCdrFile();
+        FileInfo fileInfo = new FileInfo(helper.cdr(), helper.cdrChecksum(), helper.cdrCount());
+        Map<String, Object> eventParams = new HashMap<>();
+        VerifyResults result = cdrFileService.verifyDetailFile(fileInfo);
+        assertEquals(0, result.getErrors().size());
+    }
+
+
+    @Test(expected = IllegalStateException.class)
+    public void testChecksumError() throws IOException, NoSuchAlgorithmException {
+        getLogger().debug("testChecksumError()");
+
+        CdrHelper helper = new CdrHelper(settingsService, subscriptionService, subscriberDataService,
+                languageDataService, languageLocationDataService, circleDataService, stateDataService,
+                districtDataService, fileAuditRecordDataService);
+
+        helper.makeCdrs(1, 1, 1, 1);
+        helper.makeCdrFile();
+        FileInfo fileInfo = new FileInfo(helper.cdr(), "invalid checksum", helper.cdrCount());
+        VerifyResults result = cdrFileService.verifyDetailFile(fileInfo);
+        assertEquals(0, result.getRecords().size()); //verifyDetailFile does not populate VerifyResults.records
+        assertEquals(1, result.getErrors().size());
```

Comment:
If the expectation of this test is an IllegalStateException then how come the last thing it does is an assert?  It seems like those lines wouldn't be reached.